examples/update-on-async-rendering/updating-state-from-props-before.js

```diff
@@ -0,0 +1,23 @@
+// Before
+class ExampleComponent extends React.Component {
+  // highlight-range{1-5}
+  state = {
+    derivedData: computeDerivedState(
+      this.props
+    ),
+  };
+
+  // highlight-range{1-12}
+  componentWillReceiveProps(nextProps) {
+    if (
+      this.props.someValue !==
+      nextProps.someValue
+    ) {
+      this.setState({
+        derivedData: computeDerivedState(
+          nextProps
+        ),
+      });
```

Comment:
My overall feedback on these examples: I see what you are trying to do but I think their vagueness is more distracting than helping. I would vastly prefer a specific example (that also shows when I was supposed to use this pattern in the first place).

For example, before:

```js
  state = {
    isScrollingDown: false,
  };

  componentWillReceiveProps(nextProps) {
    if (this.props.currentRow === nextProps.currentRow) {
      return;
    }
    this.setState({
      isScrollingDown: nextProps.currentRow > this.props.currentRow
    });
  }
```

After:

```js
  state = {};

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.currentRow === prevState.lastRow) {
      return null;
    }
    return {
      lastRow: nextProps.currentRow,
      isScrollingDown: nextProps.currentRow > prevState.lastRow 
    };
  }
```

Reply:
Too vague. Gotcha. I like it.